tests/test_rtdc_feat_anc_plugin.py

```diff
@@ -905,6 +964,186 @@ def test_pf_wrong_length_2():
+        assert np.allclose(events["ragged_feat"]["6"], ragged_feat[6])
+        assert events["ragged_feat"]["1"].shape == (57, 2)
+        assert events["ragged_feat"]["6"].shape == (52, 2)
+
+
+@pytest.mark.filterwarnings(
+    "ignore::dclab.rtdc_dataset.config.WrongConfigurationTypeWarning")
+def test_pf_store_image_plugin_data_with_shape():
+    """Test storage of image plugin feature attributes with shape info"""
+    h5path = retrieve_data("fmt-hdf5_fl_2018.zip")
+    info = example_plugin_info_non_scalar_feature()
+    # Mention feature shape in the info
+    info["feature shapes"] = [(80, 250)]
+    PlugInFeature("image_gauss_filter", info)
+    with dclab.new_dataset(h5path) as ds:
+        image_gauss_filter = gaussian_filter(ds["image"], sigma=(0, 1, 1))
+    with RTDCWriter(h5path) as hw:
+        hw.store_feature("image_gauss_filter", image_gauss_filter)
+    with h5py.File(h5path, mode="r") as h5:
+        events = h5["events"]
+        assert "image_gauss_filter" in events.keys()
+        feat = h5["events"]['image_gauss_filter']
+        assert feat.ndim == 3
+        fshape = feat.shape
+        assert fshape[1] > 3 and fshape[2] > 3
+        assert feat.attrs.keys()
+        assert b'CLASS' in feat.attrs.keys()
+        assert b'IMAGE_SUBCLASS' in feat.attrs.keys()
+        assert b'IMAGE' in feat.attrs.get('CLASS')
+
+
+@pytest.mark.filterwarnings(
+    "ignore::dclab.rtdc_dataset.config.WrongConfigurationTypeWarning")
+def test_pf_store_image_plugin_data_without_shape():
+    """Test storage of image plugin feature attributes without shape info"""
+    h5path = retrieve_data("fmt-hdf5_fl_2018.zip")
+    info = example_plugin_info_non_scalar_feature()
+    PlugInFeature("image_gauss_filter", info)
+    with dclab.new_dataset(h5path) as ds:
+        image_gauss_filter = gaussian_filter(ds["image"], sigma=(0, 1, 1))
+    with RTDCWriter(h5path) as hw:
+        hw.store_feature("image_gauss_filter", image_gauss_filter)
+    with h5py.File(h5path, mode="r") as h5:
+        events = h5["events"]
+        assert "image_gauss_filter" in events.keys()
+        feat = h5["events"]['image_gauss_filter']
+        assert feat.ndim == 3
+        fshape = feat.shape
+        assert fshape[1] > 3 and fshape[2] > 3
+        assert feat.attrs.keys()
+        assert b'CLASS' in feat.attrs.keys()
+        assert b'IMAGE_SUBCLASS' in feat.attrs.keys()
+        assert b'IMAGE' in feat.attrs.get('CLASS')
+
+
+@pytest.mark.filterwarnings(
+    "ignore::dclab.rtdc_dataset.config.WrongConfigurationTypeWarning")
+def test_pf_store_mask_plugin_data_with_shape():
+    """Test storage of mask plugin feature attributes with shape info"""
+    h5path = retrieve_data("fmt-hdf5_fl_2018.zip")
+    info = example_plugin_info_mask_feature()
+    # Mention feature shape in the info
+    info["feature shapes"] = [(80, 250)]
+    PlugInFeature("mask_other", info)
+    with dclab.new_dataset(h5path) as ds:
+        mask_other = np.logical_not(ds["mask"])
```

Comment:
You are not using plugin features correctly FYI. Please check out the plugin feature explanation on dclab.

Reply:
I fixed it. You can see the changes here (https://github.com/ZELLMECHANIK-DRESDEN/dclab/pull/172/commits/326e58ac6dc1df5391586c71c8c8c1b5db9f934e)